File: bsearch.py
arr = [1, 2, 3, 4, 5, 6]
x = 1

i = 0
j = len(arr) -1
while i <= j:
   if arr[int((i+j) * 0.5)] == x:
       print int((i+j) * 0.5)
       break
   elif arr[int(i+j) * 0.5)] > x:
       j = j-1
   else:
      i = i + 1

File: bsearch.cpp
#include <cstdio>
#include <cstdlib>

bool bsearch(int arr[], int n, int len)
{
     int l = 0, h = len - 1;
     int mid;
     while (l <= h)
     {
         mid = (l + h)/2;
         if (arr[mid] == n)
         {
             return true;
         }
         else if (arr[mid] < n)
         {
             l = mid + 1;
         }
         else
         {
             h = mid - 1;
         }
     }
     return false;
}


int main(int argc, char ** argv)
{
     int i, arr[argc-2];
     for(i = 1; i < argc - 1; i++)
     {
         arr[i-1] = atoi(argv[i]);
     }
        
     int n = atoi(argv[argc-1]);
     if (bsearch(arr, n, argc - 2))
     {
         printf("Found\n");
     }
     else
     {
         printf("Not found\n");
     }
     
     return 0;
} 
